x-pack/plugins/maps/server/maps_telemetry/maps_telemetry.ts

```diff
@@ -299,30 +303,58 @@ export function buildMapsTelemetry({
     },
   };
 }
-async function getMapSavedObjects(
-  savedObjectsClient: SavedObjectsClientContract | ISavedObjectsRepository
-) {
-  const mapsSavedObjects = await savedObjectsClient.find<MapSavedObjectAttributes>({
-    type: MAP_SAVED_OBJECT_TYPE,
-  });
-  return mapsSavedObjects.saved_objects;
-}
 
-async function getIndexPatternSavedObjects(
-  savedObjectsClient: SavedObjectsClientContract | ISavedObjectsRepository
+export async function execTransformOverMultipleSavedObjectPages<T>(
+  savedObjectType: string,
+  transform: (savedObjects: Array<SavedObject<T>>) => void
 ) {
-  const indexPatternSavedObjects = await savedObjectsClient.find<IndexPatternAttributes>({
-    type: 'index-pattern',
-  });
-  return indexPatternSavedObjects.saved_objects;
+  const savedObjectsClient = getInternalRepository();
+
+  let currentPage = 1;
+  // Seed values
+  let page = 1;
+  let perPage = 2;
+  let total = 3;
+  let savedObjects = [];
+
+  do {
+    const savedObjectsFindResult = await savedObjectsClient.find<T>({
+      type: savedObjectType,
+      page: currentPage++,
+    });
+    ({ page, per_page: perPage, saved_objects: savedObjects, total } = savedObjectsFindResult);
+    transform(savedObjects);
+  } while (page * perPage < total);
```

Comment:
How about just looping until `savedObjects.length < perPage`?

Reply:
It's more accurate to gauge it relative to the total. For instance if the number of maps was `40`, `savedObjects.length (20) < perPage (20)` wouldn't catch it and do another loop